client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -455,7 +463,7 @@ public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenChec
     @Override
     public void write(PendingEvent event) {
         //State is set to sealed during a Transaction abort and the segment writer should not throw an {@link IllegalStateException} in such a case.
-        checkState(NameUtils.isTransactionSegment(segmentName) || !state.isAlreadySealed(), "Segment: %s is already sealed", segmentName);
```

Comment:
We have measured this to provide some small improvement in throughput (e.g., 10K e/s), so it is good to include in this PR.